src/taco-remote-lib/common/builder.ts

```diff
@@ -128,10 +129,89 @@ class Builder {
             // Note that "cordova platform add" eventually calls "cordova prepare" internally, which is why we don't invoke prepare ourselves when we add the platform.
             return this.cordova.raw.platform("add", this.currentBuild.buildPlatform);
         } else {
-            return this.update_platform();
+            // If the client-side project has changed Cordova version or saved a new version for this platform in config.xml, then re-add the platform
+            return this.reAddPlatformIfNecessary().then((reAddedPlatform) => {
+                if (reAddedPlatform) {
+                    // Platform already prepared
+                    return Q({});
+                }
+
+                return this.update_platform();
+            });
         }
     }
 
+    /**
+     * Removes and add the platform if necessary (Cordova version changed, or config.xml has a saved platform version that differs from the installed one). Returns true if the platform was added,
+     * false if not.
+     */
+    private reAddPlatformIfNecessary(): Q.Promise<boolean> {
```

Comment:
how about we rename reAddPlatformIfNecessary to addPlatform or ensurePlatformAdded
addPlatform can also have the !fs.existsSync case inside it

AddPlatform logic could be
if (!exist) {add} else if (versionmismatch) {remove; add}